airflow/models.py

```diff
@@ -2909,23 +3008,30 @@ class DoneTrigger(Trigger):
     Trigger that matches the state of the depended upon tasks with any of the
     states returned by State.done().
     """
-    def __init__(self, task, past_executions=0):
-        Trigger.__init__(self, task, None, past_executions)
+    def __init__(self, task, past_executions=0, type=TriggerType.ALL):
+        Trigger.__init__(self, task, None, past_executions, type)
 
     @provide_session
     def evaluate(self, execution_date, session):
+        states_by_schedule = self._states_by_schedule(execution_date,
```

Comment:
we have mostly the same logic here as in the super class, except for the `state in State.done()` => couldn't we factor things a bit?